component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/MoesifClient.java

```diff
@@ -82,15 +82,28 @@ private void doRetry(String orgId, MetricEventBuilder builder) {
     public void publish(MetricEventBuilder builder) throws MetricReportingException {
         Map<String, Object> event = builder.build();
         ConcurrentHashMap<String, String> orgIDMoesifKeyMap = keyRetriever.getMoesifKeyMap();
+        ConcurrentHashMap<String, String> orgIdEnvMap = keyRetriever.getEnvMap();
+        LinkedHashMap properties = (LinkedHashMap) event.get("properties");
 
         String orgId = (String) event.get(Constants.ORGANIZATION_ID);
         String moesifKey;
+        String eventEnvironment = (String) properties.get(Constants.DEPLOYMENT_TYPE);
+        String userSelectedEnvironment;
         if (orgIDMoesifKeyMap.containsKey(orgId)) {
             moesifKey = orgIDMoesifKeyMap.get(orgId);
+            if (orgIdEnvMap.containsKey(orgId)) {
+                userSelectedEnvironment = orgIdEnvMap.get(orgId);
+            } else {
+                return;
+            }
         } else {
             return;
         }
 
+        if (userSelectedEnvironment.equals(Constants.PRODUCTION) && !eventEnvironment.equals(Constants.PRODUCTION)) {
```

Comment:
Do the equal comparison against the constant, i.e Constants.PRODUCTION.equals(userSelectedEnvironment). This prevents NPEs.

Reply:
Resolved with: 9a0b30dc39ce6c0986d843daf6803adebbe92e56